rpc, kvserver: unskip `TestAdminScatterWithDrainingNodes` test

cthumuluru-crdb opened the pull request:
This patch re-enables the `TestAdminScatterWithDrainingNodes` test, which had been skipped when running in drpc mode due to an underlying issue that (involves drpc soft cancel option) is now resolved.

Soft cancel had previously been enabled by default to mitigate connection churn caused by stream pooling. However, a bug in the soft cancel implementation causes the connection reader to hang indefinitely, preventing new server streams from being created. Until a proper fix is available, this patch disables soft cancel by default.

For additional context, see the discussion in ticket https://github.com/cockroachdb/cockroach/issues/158270 and PR https://github.com/cockroachdb/cockroach/pull/145907.

Fixes: #158270, #158524
Epic: CRDB-55382
Release note: None

tbg reviewed (approved):
LGTM mod the personal fork etc which you're already tracking via the `do-not-merge` label.

> @tbg reviewed 1 of 1 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/158962#-:-Og6cF83EDX6d6knHhCv:b-jeo960)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @cthumuluru-crdb)



<!-- Sent from Reviewable.io -->

cthumuluru-crdb committed:
kvserver: unskip TestAdminScatterWithDrainingNodes test

This patch re-enables the `TestAdminScatterWithDrainingNodes` test,
which had been skipped when running in drpc mode due to an underlying
issue that (involves drpc soft cancel option) is now resolved.

Soft cancel had previously been enabled by default to mitigate
connection churn caused by stream pooling. However, a bug in the soft
cancel implementation causes the connection reader to hang indefinitely,
preventing new server streams from being created. Until a proper fix is
available, this patch disables soft cancel by default.

For additional context, see the discussion in ticket
https://github.com/cockroachdb/cockroach/issues/158270 and
PR https://github.com/cockroachdb/cockroach/pull/145907.

Fixes: 158270
Epic: CRDB-55382
Release note: None

Nukitt reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

side note: iirc soft-cancel was also the culprit in https://github.com/cockroachdb/cockroach/issues/158524, maybe we can re-enable that too?

> @Nukitt reviewed 1 of 5 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/158962#-:-Og7J1xTDuVIsusby334:bikdwja)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @shubhamdhama and @suj-krishnan)



<!-- Sent from Reviewable.io -->
